File: VB/CustomFunctionForExpressionEditorExample/Functions/CustomReportExpressionEditorCustomizationService.vb
Imports DevExpress.Data.Controls.ExpressionEditor
Imports DevExpress.XtraReports.UserDesigner

Public Class CustomReportExpressionEditorCustomizationService
	Inherits ReportExpressionEditorCustomizationService

	Public Sub New()
	End Sub
	Public Overrides Sub BeforeRun(ByVal expressionString As String,
								   ByVal expressionEditorView As IExpressionEditorView,
								   ByVal expressionEditorContext As ExpressionEditorContext)
		Dim formatStringFunc As FunctionInfo =
			expressionEditorContext.Functions.FirstOrDefault(Function(x) x.Name = "Abs")
		expressionEditorContext.Functions.Remove(formatStringFunc)
		MyBase.BeforeRun(expressionString, expressionEditorView, expressionEditorContext)
	End Sub
End Class

File: CS/CustomFunctionForExpressionEditorExample/Functions/CustomReportExpressionEditorCustomizationService.cs
using DevExpress.Data.Controls.ExpressionEditor;
using DevExpress.XtraReports.UserDesigner;
using System.Linq;

namespace CustomFunctionForExpressionEditorExample
{
    class CustomReportExpressionEditorCustomizationService : ReportExpressionEditorCustomizationService
    {
        public CustomReportExpressionEditorCustomizationService() {}
        public override void BeforeRun(string expressionString, 
            IExpressionEditorView expressionEditorView, 
            ExpressionEditorContext expressionEditorContext)
        {
            FunctionInfo formatStringFunc = expressionEditorContext.Functions
                .FirstOrDefault(x => x.Name == "Abs");
            expressionEditorContext.Functions.Remove(formatStringFunc);
            base.BeforeRun(expressionString, expressionEditorView, expressionEditorContext);
        }
    }
}
